pkg/upload/unity-ios.go

```diff
@@ -0,0 +1,156 @@
+package upload
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bugsnag/bugsnag-cli/pkg/ios"
+	"github.com/bugsnag/bugsnag-cli/pkg/log"
+	"github.com/bugsnag/bugsnag-cli/pkg/options"
+	"github.com/bugsnag/bugsnag-cli/pkg/unity"
+	"github.com/bugsnag/bugsnag-cli/pkg/utils"
+)
+
+func ProcessUnityIos(globalOptions options.CLI, endpoint string, logger log.Logger) error {
+	var (
+		err              error
+		possibleDsymPath string
+		dsyms            []*ios.DwarfInfo
+		tempDir          string
+		tempDirs         []string
+		lineMappingFile  string
+		plistData        *ios.PlistData
+	)
+
+	unityOptions := globalOptions.Upload.UnityIos
+
+	defer func() {
+		for _, dir := range tempDirs {
+			_ = os.RemoveAll(dir)
+		}
+	}()
+
+	for _, path := range unityOptions.Path {
+		if unityOptions.DsymShared.Scheme == "" {
+			unityOptions.DsymShared.Scheme = "Unity-iPhone"
+			logger.Debug(fmt.Sprintf("Using default Unity scheme: %s", unityOptions.DsymShared.Scheme))
+		}
+
+		if ios.IsPathAnXcodeProjectOrWorkspace(path) {
+			globalOptions.Upload.XcodeArchive = options.XcodeArchive{
+				Path:   utils.Paths{path},
+				Shared: unityOptions.DsymShared,
+			}
+
+			xcarchivePath, err := ios.FindXcarchivePath(globalOptions, logger)
+			if err != nil {
+				return fmt.Errorf("failed to find Xcode archive path: %w", err)
+			}
+			if xcarchivePath == "" {
+				return fmt.Errorf("no Xcode archive found in specified paths")
+			}
+
+			logger.Info(fmt.Sprintf("Found Xcode archive at %s", xcarchivePath))
+			possibleDsymPath = xcarchivePath
+		} else {
+			possibleDsymPath = path
+		}
+
+		dsyms, tempDir, err = ios.FindDsymsInPath(
+			possibleDsymPath,
+			unityOptions.DsymShared.IgnoreEmptyDsym,
+			unityOptions.DsymShared.IgnoreMissingDwarf,
+			logger,
+		)
+
+		tempDirs = append(tempDirs, tempDir)
+
+		if err != nil {
+			return fmt.Errorf("error locating dSYMs in %s: %w", possibleDsymPath, err)
+		}
+
+		if len(dsyms) == 0 {
+			return fmt.Errorf("no dSYMs found in %s", possibleDsymPath)
+		}
+
+		logger.Info(fmt.Sprintf("Found %d dSYM files in: %s", len(dsyms), possibleDsymPath))
+
+		if globalOptions.Upload.UnityIos.VersionName == "" || globalOptions.Upload.UnityIos.BundleVersion == "" || globalOptions.Upload.UnityIos.ApplicationId == "" {
+			plistPath := string(unityOptions.DsymShared.Plist)
+			if plistPath == "" {
+				plistPath = filepath.Join(dsyms[0].Location, "..", "..", "Info.plist")
+			}
+
+			logger.Debug(fmt.Sprintf("Using plist path: %s", plistPath))
```

Comment:
This implies it worked - how about moving the log message inside the `if plistData != nil` check: logged that we _have_read it or else that no plist was found (info only, not an error)